PHPStan check fails on PHP 8

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 3.0.x

#4523 fixes PHPStan issues on `2.12.x` but there are some more on `3.0.x` and newer branches:

```
$ php -v
PHP 8.0.3 (cli) (built: Mar  6 2021 10:06:45) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.3, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.3, Copyright (c), by Zend Technologies
    with Xdebug v3.0.3, Copyright (c) 2002-2021, by Derick Rethans

$ git rev-parse HEAD
f18aee0a2b209662370de14f3a5268553c9b239e

$ phpstan --version
PHPStan - PHP Static Analysis Tool 0.12.80

$ phpstan a

Note: Using configuration file /Users/smorozov/Projects/dbal/phpstan.neon.dist.
 258/258 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ -------------------------------------------------------------------------------------
  Line   Driver/Mysqli/Result.php
 ------ -------------------------------------------------------------------------------------
  135    Call to function assert() with true will always evaluate to true.
  135    Call to function is_array() with array<string, mixed> will always evaluate to true.
 ------ -------------------------------------------------------------------------------------

 ------ ---------------------------------------------------------------------------------
  Line   Driver/OCI8/Result.php
 ------ ---------------------------------------------------------------------------------
  101    Strict comparison using !== between int and false will always evaluate to true.
  105    Unreachable statement - code above always terminates.
 ------ ---------------------------------------------------------------------------------

 ------ -----------------------------------------------------------------------------------
  Line   Driver/OCI8/Statement.php
 ------ -----------------------------------------------------------------------------------
  102    Call to function assert() with true will always evaluate to true.
  102    Strict comparison using !== between OCILob|null and false will always evaluate to
         true.
 ------ -----------------------------------------------------------------------------------

 [ERROR] Found 6 errors
```
The same build passes on PHP 7.4.

--- a/.github/workflows/static-analysis.yml
+++ b/.github/workflows/static-analysis.yml
@@ -16,7 +16,7 @@ jobs:
     strategy:
       matrix:
         php-version:
-          - "7.4"
+          - "8.0"
 
     steps:
       - name: "Checkout code"

--- a/phpstan.neon.dist
+++ b/phpstan.neon.dist
@@ -29,7 +29,7 @@ parameters:
 
         # weird class name, represented in stubs as OCI_(Lob|Collection)
         - '~unknown class OCI-(Lob|Collection)~'
-        - '~^Call to method writeTemporary\(\) on an unknown class OCILob\.~'
+        - '~^Call to method writetemporary\(\) on an unknown class OCILob\.~'
 
         # Requires a release of https://github.com/JetBrains/phpstorm-stubs/pull/553
         -
@@ -107,5 +107,23 @@ parameters:
             message: '~^Instanceof between Doctrine\\DBAL\\Platforms\\Keywords\\KeywordList and Doctrine\\DBAL\\Platforms\\Keywords\\KeywordList will always evaluate to true\.~'
             paths:
                 - %currentWorkingDirectory%/src/Platforms/AbstractPlatform.php
+
+        # TODO: remove this once the support for PHP 7 is dropped
+        -
+            message: '~^Strict comparison using !== between int and false will always evaluate to true\.$~'
+            paths:
+                - %currentWorkingDirectory%/src/Driver/OCI8/Result.php
+        -
+            message: '~^Unreachable statement - code above always terminates\.$~'
+            paths:
+                - %currentWorkingDirectory%/src/Driver/OCI8/Result.php
+        -
+            message: '~^Call to function assert\(\) with true will always evaluate to true\.$~'
+            paths:
+                - %currentWorkingDirectory%/src/Driver/OCI8/Statement.php
+        -
+            message: '~^Strict comparison using !== between OCILob\|null and false will always evaluate to true\.$~'
+            paths:
+                - %currentWorkingDirectory%/src/Driver/OCI8/Statement.php
 includes:
     - vendor/phpstan/phpstan-strict-rules/rules.neon

--- a/src/Driver/Mysqli/Result.php
+++ b/src/Driver/Mysqli/Result.php
@@ -131,10 +131,7 @@ final class Result implements ResultInterface
             return false;
         }
 
-        $row = array_combine($this->columnNames, $values);
-        assert(is_array($row));
-
-        return $row;
+        return array_combine($this->columnNames, $values);
     }
 
     /**

